src/main/java/at/hannibal2/skyhanni/features/misc/HideExCoopMembers.kt

```diff
@@ -0,0 +1,184 @@
+import at.hannibal2.skyhanni.utils.StringUtils.cleanPlayerName
+import at.hannibal2.skyhanni.utils.StringUtils.isPlayerName
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.init.Items
+
+@SkyHanniModule
+object HideExCoopMembers {
+
+    private val config get() = SkyHanniMod.feature.misc
+    private val storage get() = ProfileStorageData.profileSpecific
+
+    private val historicMembersInventory = InventoryDetector { name -> inventoryPattern.matches(name) }
+    private val collectionInventory = InventoryDetector { name -> collectionInventoryPattern.matches(name) }
+
+    private var changedSlotNumber: Int? = null
+
+    private val patternGroup = RepoPattern.group("data.hiddenmembers")
+
+    /**
+     * REGEX-TEST: Historic Members
+     */
+    private val inventoryPattern by patternGroup.pattern(
+        "inventory.historic",
+        "Historic Members",
+    )
+
+    /**
+     * REGEX-TEST: Farming Collections
+     * REGEX-TEST: Carrot Collection
+     */
+    private val collectionInventoryPattern by patternGroup.pattern(
+        "inventory.collections",
+        ".+ Collections?",
+    )
+
+    /**
+     * REGEX-TEST: §b[MVP§f+§b] oxsss§7: §e1.9M
+     * REGEX-TEST: §a[VIP] oxsss§7: §e0
+     * REGEX-TEST: §7oxsss§7: §e0
+     */
+    private val collectedPattern by patternGroup.pattern(
+        "inventory.collections.collected",
+        "(?:§.\\[[^]]+(?:§\\++§b)?] |§7)(?<name>[^§]{2,16})§7: §e(?<amount>.+)",
+    )
+
+    /**
+     * REGEX-TEST: §7Progress to Raw Chicken IX: §e25.7§6%
+     * REGEX-TEST: §7Total Collected: §e1,917,287
+     */
+    private val dontDisplayMaxedPattern by patternGroup.pattern(
+        "inventory.collections.displaymaxed",
+        "§7(?:Progress to .+|Total Collected: .+)",
+    )
+
+    @HandleEvent
+    fun onTooltip(event: ToolTipEvent) {
+        if (!config.hideExCoopMembers || !collectionInventory.isInside()) return
+        val hiddenMembers = storage?.hiddenCoopMembers.takeIf { !it.isNullOrEmpty() } ?: return
+
+        event.toolTip = event.toolTipRemovedPrefix().handleTooltip(hiddenMembers)
+        changedSlotNumber = event.slot.slotNumber
+    }
+
+    private fun List<String>.handleTooltip(storage: MutableSet<String>): MutableList<String> = this.toMutableList().apply {
+        val coopIndex = indexOf("§7Co-op Contributions:")
+        if (coopIndex == -1) return@apply
+
+        var remainingPlayers = 0
+        var totalCollected = 0
+        val itemsToRemove = mutableListOf<Int>()
+
+        drop(coopIndex).forEachIndexed { index, line ->
+            if (line.isBlank()) return@forEachIndexed
+
+            collectedPattern.matchMatcher(line) {
+                if (group("name") in storage) {
+                    itemsToRemove.add(coopIndex + index)
+                } else {
+                    remainingPlayers++
+                }
+
+                totalCollected += group("amount").formatInt()
```

Comment:
i have already mentioned the same thing above:we have an collection api. please use it. if the exact data you need dont exist in the collection api, please add them to the collection api instead of adding code duplication. points for reference include: https://github.com/hannibal002/SkyHanni/blob/b60151d0f7b6b298caea8d238e4d096eab7f7907/src/main/java/at/hannibal2/skyhanni/api/CollectionApi.kt#L78

and 

https://github.com/hannibal002/SkyHanni/blob/ff837769c75b2f70b35f1a11779d594d2e1a35fb/src/main/java/at/hannibal2/skyhanni/features/inventory/ItemDisplayOverlayFeatures.kt#L212